modules/vmfcore/src/reader_base.cpp

```diff
@@ -0,0 +1,38 @@
+/* 
+ * Copyright 2015 Intel(r) Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http ://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "vmf/reader_base.hpp"
+
+namespace vmf
+{
+
+std::string ReaderBase::decompress(const std::string& input)
+{
+    std::string decompressed;
+    if(compressor)
+    {
+        vmf_rawbuffer compressed(input.c_str(), input.size());
```

Comment:
* base64 encoding/decoding in compression implemented
* now `compressorId` is used instead of shared_ptr<Compressor>